### PR TITLE
docs: key registry documentation

### LINKED ISSUE #493: Document the Key Registry

### PR DESCRIPTION
**Pull request by njlie**
## Changes proposed in this pull request

- Adds documentation for how a Key Registry should be implemented, and how it is used by an AS.

## Context

Fixes #493.

## Checklist

- [x] Related issues linked using `fixes #number`
- [ ] Tests added/updated
- [x] Documentation added
- [ ] Make sure that all checks pass

### CODE REVIEW

File: docs/key-registry.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Key Registry
+
+## Sequence Diagram
+
+TODO: add sequence diagram
```
Comment (sabineschaller) [inline docs/key-registry.md:5]: Are you planning to do this in this PR?

File: docs/key-registry.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+## Basics
+
+### What is the key registry?
+
+A key registry is a list of keys stored by a client. A key is generated and added to the registry by a client and exposed publically at the path `/jwks.json` in anticipation of a grant request on an authorization server (AS).
```
Comment (sabineschaller) [inline docs/key-registry.md:11]: Maybe elaborate on what a client is? I think we should at least mention that a client is represented by a payment pointer and that the keys are at the path `PAYMENT_POINTER/jwks.json`

File: docs/key-registry.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+### What is the purpose of the key registry?
+
+The key registry allows an AS to verify that a client is who they say they are. Because a grant request is completed over multiple HTTP requests it is thus important for a client to provide a way to consistently identify itself across these requests to the AS.
```
Comment (sabineschaller) [inline docs/key-registry.md:15]: Maybe mention that requests are signed here already?

File: docs/key-registry.md (hunk 1 above)
Comment (njlie) [reply docs/key-registry.md:5]: I actually have one ready to go, just forgot to add it.

File: docs/key-registry.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
+# Key Registry
```
Comment (wilsonianb) [inline docs/key-registry.md:1]: I'm not convinced this needs its own doc or that key registry is a meaningful concept given how baked in it is to Open Payments :grimacing: 

Open Payments docs already describe:
- https://docs.openpayments.guide/reference/get-payment-pointer-keys
- https://docs.openpayments.guide/reference/post-request `client` field
- https://docs.openpayments.guide/reference/create-incoming-payment, etc `SIGNATURE_INPUT`

and Rafiki admin docs include:
- https://rafiki.dev/docs/reference/mutations#createpaymentpointerkey
- https://rafiki.dev/docs/reference/mutations#revokepaymentpointerkey

Comment (sabineschaller) [reply docs/key-registry.md:1]: We'll sort through docs next week. I rather have it here than not have it at all. We may just move it somewhere else next week.

File: docs/key-registry.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+## Sequence Diagram
+
+![key-registry-diagram](https://user-images.githubusercontent.com/20246798/229255746-290a5a54-1f1e-4c7c-b891-4a27152c743e.png)
```
Comment (sabineschaller) [inline docs/key-registry.md:5]: Could you also put the source code somewhere?

**Review by sabineschaller — APPROVED** [review]

**Review by sabineschaller — APPROVED** [review]

**Pull request merged**